src/Microsoft.ML.Transforms/NAReplaceTransform.cs

```diff
@@ -165,6 +167,44 @@ private static string TestType<T>(ColumnType type)
             return null;
         }
 
+        public class ColumnInfo
+        {
+            public readonly string Input;
+            public readonly string Output;
+            public readonly bool ImputeBySlot;
+            public readonly ReplacementKind Kind;
+
+            public ColumnInfo(string input, string output, ReplacementKind kind = ReplacementKind.DefaultValue, bool imputeBySlot = true)
+            {
+                Input = input;
+                Output = output;
+                ImputeBySlot = imputeBySlot;
+                Kind = kind;
+            }
+            internal string ReplacementString { get; set; }
+        }
+
+        private static (string input, string output)[] GetColumnPairs(ColumnInfo[] columns)
+        {
+            Contracts.CheckValue(columns, nameof(columns));
+            return columns.Select(x => (x.Input, x.Output)).ToArray();
+        }
+
+        ///IVAN: move to mapper.
```

Comment:
>move to mapper. [](start = 17, length = 15)

maybe just turn into one 'type' ? The rest is accessible via `_parent.ColumnPairs`

Reply:
I'm actually quite tempted to move whole ColInfo into MapperBase.
Since it's heavily used almost in every mapper I wrote.

But yeah, I have _types array which I will use in Estimator.

---
In reply to: [217834859](https://github.com/dotnet/machinelearning/pull/917#discussion_r217834859) [](ancestors = 217834859)

Reply:
Moving `ColInfo` to `MapperBase` will be a bit unwieldy I think.

---
In reply to: [217836517](https://github.com/dotnet/machinelearning/pull/917#discussion_r217836517) [](ancestors = 217836517,217834859)